kubeweekly/values.yaml

```diff
@@ -1,23 +1,38 @@
----
-# default values supplied for templates/* files
-job:
-  name: example-app
-  namespace: default
-  image: "brndnmtthws/nginx-echo-headers"
-  schedule: "*/12 * * * 3"
-  initContainers:
-    name: kubeweekly-scrapper
-    image: kubeweekly-scrapper:latest
-    args: npm run trigger
+jobLimit: 1
 
+scrapper:
+  name: kubeweekly-scrapper-jobs
+  namespace: kubeweekly
+  image: "cloudnativeid/kubeweekly:1.0.0"
+  schedule: "0 12 * * *"
+  args: "npm run trigger"
   env:
-    - name: TOKEN
-      valueFrom:
-        secretKeyRef:
-          name: example-app-secrets
-          key: telegram-bot-token
-    - name: CHATID
-      valueFrom:
-        secretKeyRef:
-          name: example-app-secrets
-          key: telegram-bot-chatid
+  - name: USER
+    value: 'cncfidbot'
+  - name: PASS
+    valueFrom:
+      secretKeyRef:
+        name: github-token-secret
```

Comment:
how did you create this secret? ideally the creation should also be included in this helm chart repo, but dont commit the value (maybe through env vars), check if there's easy way to pass secret in argocd

Reply:
I create secret manually from kubectl, I think we can change with https://github.com/bitnami-labs/sealed-secrets

Reply:
I think it's impossible to create a secret directly trough the Helm chart because we need to store the raw key in values.yaml.

I cannot find the environment variable mechanism in the `Application` kind argoCD. If we can, we still store the key inside apps.yaml.

I think the best way is `git secret` (only accessible from GitHub actions), can we make predefine sealed-secrets from GitHub Actions so argoCD just apply into the cluster?

Reply:
sure git secret works as well, then you add another step to do `envsubst` values.yaml or manifests in the CI (github actions)

Reply:
sure git secret works as well, then you add another step to do `envsubst` values.yaml or manifests in the CI (github actions)

Reply:
Now I separate the secret files from charts, should I merge the secret?

for now, we have `secret.tmpl` that will generate by running `envsubst` and apply into the cluster before running apps.yaml. (check gke.yml actions)

should we merge the secret into the charts?

Reply:
> Now I separate the secret files from charts, should I merge the secret?
> 
> for now, we have `secret.tmpl` that will generate by running `envsubst` and apply into the cluster before running apps.yaml. (check gke.yml actions)
> 
> should we merge the secret into the charts?

Hm, I think this is will not work since argoCD always running with a 3-minute interval. and values.yaml cannot be populated from argoCD.